samcli/lib/build/bundler.py

```diff
@@ -32,6 +36,12 @@ def esbuild_configured(self) -> bool:
                 return True
         return False
 
+    def handle_template_post_processing(self) -> Dict:
+        template = deepcopy(self._previous_template)
```

Comment:
Idea: If there are multiple deepcopies , there might be something to just copy once and set it in thread local storage.

Reply:
There should only be two deepcopies, one for the pre-processed template and one for the post-processed template which should be different templates.